### PR TITLE
[TR] PIM-3338: Mass-review proposals

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-3338

Generalization of the type of batch step used in EE for mass publish / unpublish. The goal is to reuse it for mass review.

| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | Y
| BC breaks?           | N
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | -
| Checkstyle issues?*  | -
| PMD issues?**        | -
| Changelog updated?   | -
| Fixed tickets        | -
| DB schema updated?   | N
| Migration script?    | -
| Doc PR               |

EE PR : https://github.com/akeneo/pim-enterprise-dev/pull/852

### CODE REVIEW

File: src/Pim/Bundle/BaseConnectorBundle/Step/TaskletInterface.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Pim\Bundle\BaseConnectorBundle\Step;
+
+use Akeneo\Bundle\BatchBundle\Step\StepExecutionAwareInterface;
+
+/**
+ * Simple task to be executed from a TaskletStep.
+ *
+ * @author Yohan Blain <yohan.blain@akeneo.com>
```
Comment (nidup) [inline src/Pim/Bundle/BaseConnectorBundle/Step/TaskletInterface.php]: you can add licence and copyright in CE

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/steps.yml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 parameters:
+    pim_base_connector.step.tasklet.class: Pim\Bundle\BaseConnectorBundle\Step\TaskletStep
```
Comment (jmleroux) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/steps.yml]: alignment

File: src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public function getIdentifierProperties()
+    {
+        return ['username', 'email'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findOneByIdentifier($identifier)
```
Comment (sumbobyboys) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php]: weird that we don't already have a method for that

Comment (phaseinducer) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php]: I think we use that until now : https://github.com/akeneo/platform/blob/pim-1.4/src/Oro/Bundle/UserBundle/Entity/UserManager.php#L114

File: src/Pim/Bundle/BaseConnectorBundle/Step/TaskletStep.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Pim\Bundle\BaseConnectorBundle\Step;
+
+use Akeneo\Bundle\BatchBundle\Entity\StepExecution;
+use Akeneo\Bundle\BatchBundle\Step\AbstractStep;
+
+/**
+ * @author    Adrien Pétremann <adrien.petremann@akeneo.com>
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Step/TaskletStep.php]: sure ?

File: src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        $qb = $this->createQueryBuilder('u');
+        $qb->where($qb->expr()->orX(
```
Comment (skeleton) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php]: IMHO it's very complicated to read that, whereas it's just a simple SQL request

File: src/Pim/Bundle/BaseConnectorBundle/Step/TaskletStep.php (hunk 4 above)
Comment (phaseinducer) [inline src/Pim/Bundle/BaseConnectorBundle/Step/TaskletStep.php]: Actually I moved it from EE (https://github.com/akeneo/pim-enterprise-dev/blob/e1ba59a6092be9704773fef2a5f7579c00c43378/src/PimEnterprise/Bundle/BaseConnectorBundle/Step/MassEditProductPublicationStep.php), @grena wrote it. ; )

File: src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php (hunk 5 above)
Comment (phaseinducer) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php]: Ok, is there a better way to do that ?

Comment (skeleton) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepository.php]: `$qb->where('u.username = :identifier OR u.email = :identifier')` ?

File: src/Pim/Bundle/UserBundle/Entity/Repository/UserRepositoryInterface.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+use Doctrine\Common\Persistence\ObjectRepository;
+
+/**
+ * User repository interface
+ *
+ * @author    Yohan Blain <yohan.blain@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface UserRepositoryInterface extends IdentifiableObjectRepositoryInterface, ObjectRepository
```
Comment (nidup) [inline src/Pim/Bundle/UserBundle/Entity/Repository/UserRepositoryInterface.php]: +1

**Comment by nidup** [comment]
GTM

**Pull request merged**